authors/apps/articles/views.py

```diff
@@ -156,41 +168,116 @@ def create(self, request, article_slug=None):
         filters = {
             self.lookup_field: self.kwargs[self.lookup_url_kwarg]
         }
         comments = self.queryset.filter(**filters)
-        serializer = CommentSerializer(comments, many=True)
-        data = serializer.data
+        whole, sections = CommentsListAPIView.groups(comments, article)
+        total_len = len(whole) + len(sections)
         return Response({
-            'comments': data,
-            'commentsCount': len(data)
-        }, status=status.HTTP_200_OK) if len(data) else \
+            'comments': whole,
+            'highlights': sections,
+            'commentsCount': total_len
+        }, status=status.HTTP_200_OK) if total_len else \
             Response({
                 'message': 'No comments found for this article'
             })
 
+    @staticmethod
+    def groups(comments, article):
+        """Group the comments into highlights and full article comment"""
+        sections = []
+        whole = []
+        for comment in comments:
+            if all([comment.begin == 0, comment.end == -1]):
+                whole.append(CommentSerializer(comment).data)
+                continue
+            one = CommentSerializer(comment).data
+            ranges = {"start": comment.begin}
+            ranges.update({"end": comment.end})
+            one.update({"selection": {"range": ranges,
+                                      "text": article.body[comment.begin:comment.end + 1]}})
+            sections.append(one)
+        return whole, sections
+
+    @staticmethod
+    def check_out_of_range(current_response, start, end, article_len):
+        response = None
+        if max(start, end) >= article_len:
+            response = [current_response, Response(dict(errors=dict(
+                range="Bad selection. "
+                      "The start and end of selection must "
+                      "be between 0 and %d inclusive" % (article_len - 1)
+            )), status=status.HTTP_400_BAD_REQUEST)][current_response is None]
+        return response
+
+    @staticmethod
+    def assign_range(response, context, start, end, range_selected):
+        if response is None:
+            context['start'] = [0, start][start > -1]
+            context['end'] = [-1, end][end > -1]
+            return
+        if len(range_selected) == 0:
+            context['start'] = 0
+            context['end'] = -1
+
+    def try_save(self, response, comment_data, context, ranges):
+        """Try to save a comment to the database"""
+        try:
+            serializer = self.serializer_class(data=comment_data, context=context)
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+            current = serializer.data
+            current.update({
+                               "Comment Type": "Highlight",
+                               "begin": ranges[0],
+                               "end": ranges[1]} if len(ranges[2]) == 2 else {
+                "Comment-Type": "Whole"
```

Comment:
Why not just Comment, if a comment can either be a Highlight or an actual comment? Try to use natural worded descriptions as much as possible